packages/idyll-compiler/src/transformers/make-full-width.js

```diff
@@ -0,0 +1,94 @@
+const {
+  DERIVED,
+  VAR,
+  DATA,
+  VALUE,
+  VARIABLE,
+  EXPRESSION,
+  createComponentNode,
+  getChildren,
+  getNodeName,
+  isTextNode,
+  getProperty,
+  hasProperty,
+  setValueProperty,
+  setExpressionProperty,
+  removeProperty
+} = require('idyll-ast');
+
+const TEXT_CONTAINER = 'TextContainer';
+const FULLWIDTH = 'fullwidth';
+const SCROLLER = 'scroller';
+const CLASS_NAME = 'className';
+
+function makeFullWidth(ast) {
+  let textNodes = [];
+  const reduced = [];
+
+  for (const child of getChildren(ast)) {
+    try {
+      const ist = isTextNode(child);
+    } catch (err) {
+      console.error(err);
+      console.error('AST', JSON.stringify(ast, 0, 2));
+    }
+    if (isTextNode(child)) {
+      textNodes.push(child);
+      continue;
+    }
+
+    const childName = getNodeName(child).toLowerCase();
+
+    if (
+      [DERIVED, VAR, DATA, FULLWIDTH, SCROLLER].includes(childName) ||
+      hasProperty(child, 'fullWidth')
+    ) {
+      if (childName === FULLWIDTH) {
+        child.name = 'div'; // TODO: use API?
```

Comment:
Flagging this `TODO` comment in case there was meant to be an update.

Reply:
Good catch. There are no AST API calls for modifications such as changing the component type. The TODO here is a reminder to come back and consider if there should be. After working on this PR further, this seems so rare that I'm thinking we can leave this as-is and remove the comment.

Reply:
Cool, makes sense to me